### PR TITLE
Add project resource type

### PR DESCRIPTION
**Pull request by ksamoray**

### CODE REVIEW

File: nsxt/resource_nsxt_policy_project.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+				Optional: true,
+				Computed: true,
+			},
+			"site_info": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"edge_cluster_paths": {
+							Type: schema.TypeList,
+							Elem: &schema.Schema{
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_project.go:46]: we can use getElemPolicyPathSchema here?

File: nsxt/resource_nsxt_policy_project.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+						},
+						"site_path": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+				Optional: true,
+			},
+			"tier0s": {
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_project.go:59]: `tier0_gateway_paths` for consistency?

File: nsxt/resource_nsxt_policy_project.go (hunk 1 above)
Comment (ksamoray) [reply nsxt/resource_nsxt_policy_project.go]: Also for the T0 path?

File: nsxt/data_source_nsxt_policy_project.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+)
+
+const defaultOrgID = "default"
+
+func dataSourceNsxtPolicyProject() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtPolicyProjectRead,
+
+		Schema: map[string]*schema.Schema{
+			"id":           getDataSourceIDSchema(),
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_project.go:22]: should this be `nsx_id` (getNsxIDSchema) as in other resources?,

File: nsxt/resource_nsxt_policy_project.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"site_info": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"edge_cluster_paths": getElemPolicyPathSchemaWithFlags(true, false, false),
+						"site_path":          getElemPolicyPathSchemaWithFlags(true, false, false),
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_project.go:43]: Is this a list or a single element?

Comment (ksamoray) [reply nsxt/resource_nsxt_policy_project.go:43]: edge_cluster_paths should be a list indeed

File: nsxt/resource_nsxt_policy_project.go [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+func resourceNsxtPolicyProjectUpdate(d *schema.ResourceData, m interface{}) error {
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining Project ID")
+	}
+
+	headers := map[string]string{"enable_partial_patch": "true"}
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_project.go:185]: Is this header still needed?

Comment (ksamoray) [reply nsxt/resource_nsxt_policy_project.go]: In theory we do: I've added this to handle that issue, where NSX was overriding short_id on update.
The NSX bug is fixed but there are still released NSX versions which support projects but do not have this fix.
However - tests seem to pass without it - so I'll remove it :)

File: nsxt/resource_nsxt_policy_project.go [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+			edgeClusterPaths = append(edgeClusterPaths, ecp.(string))
+		}
+		sitePath := data["site_path"].(string)
+		obj := model.SiteInfo{
+			EdgeClusterPaths: edgeClusterPaths,
+			SitePath:         &sitePath,
+		}
+		siteInfos = append(siteInfos, obj)
+	}
+	var tier0s []string
```
Comment (annakhm) [inline nsxt/resource_nsxt_policy_project.go:97]: Lets use `getStringListFromSchemaList` helper here?

File: nsxt/resource_nsxt_policy_project.go (hunk 5 above)
Comment (annakhm) [reply nsxt/resource_nsxt_policy_project.go]: We'll officially claim multitenancy support starting from next release, so the fix will be in

Comment (ksamoray) [reply nsxt/resource_nsxt_policy_project.go]: Cool, I'll leave `getPolicyConnectorWithHeaders()` just in case we'll even need it?

**Review by annakhm — APPROVED** [review]

**Pull request merged**